src/Microsoft.ApplicationInsights.AspNetCore/Logging/Implementation/ApplicationInsightsLoggerCallbacks.cs

```diff
@@ -0,0 +1,38 @@
+﻿namespace Microsoft.ApplicationInsights.AspNetCore.Logging
+{
+    using System;
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// Class to control default debug logger and disable it if logger was added to <see cref="ILoggerFactory"/> explicetely.
+    /// </summary>
+    internal class ApplicationInsightsLoggerCallbacks
+    {
+        /// <summary>
+        /// Stores list of callbacks to execute when new ApplicationInsights logger is added.
+        /// </summary>
+        private readonly List<Action> callbacks;
```

Comment:
You just re-invented MulticastDelegate, Just use an event:
```C#
event EventHandler LoggerAdded;
```
and let clients subscribe.